src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+            # Checking for the length of the sliced table to be greater than the ``chunk_size``
+            # since it can be smaller if the last two chunks are faulty. Note: The two last chunks
+            # can be overlapping during the first pass, so we simply ignore them if there are faulty.
+            if len(table_sliced) > aggregator.chunk_size:
+                aggregated_stats_secondpass.append(
+                    aggregator(
+                        table=table_sliced,
+                        masked_pixels_of_sample=masked_pixels_of_sample,
+                        col_name=col_name,
+                        chunk_shift=self.chunk_shift,
+                    )
+                )
+        # Stack the aggregated statistics of each faulty chunk
+        aggregated_stats_secondpass = vstack(aggregated_stats_secondpass)
+        # Detect faulty pixels with multiple instances of OutlierDetector of the second pass
+        outlier_mask_secondpass = np.zeros_like(
+            aggregated_stats_secondpass["mean"], dtype=bool
+        )
+        for (
+            aggregated_val,
+            outlier_detector,
+        ) in self.outlier_detectors.items():
+            outlier_mask_secondpass = np.logical_or(
+                outlier_mask_secondpass,
+                outlier_detector(aggregated_stats_secondpass[aggregated_val]),
+            )
+        # Add the outlier mask to the aggregated statistics
+        aggregated_stats_secondpass["outlier_mask"] = outlier_mask_secondpass
+        aggregated_stats_secondpass["is_valid"] = self._get_valid_chunks(
+            outlier_mask_secondpass
+        )
+        return aggregated_stats_secondpass
+
+    def _get_valid_chunks(self, outlier_mask):
+        """
+        Identify valid chunks based on the outlier mask.
+
+        This method processes the outlier mask to determine which chunks of data
+        are considered valid or faulty. A chunk is marked as faulty if the percentage
+        of outlier pixels exceeds a predefined threshold ``faulty_pixels_threshold``.
+
+        Parameters
+        ----------
+        outlier_mask : numpy.ndarray
+            Boolean array indicating outlier pixels. The shape of the array should
+            match the shape of the aggregated statistics.
+
+        Returns
+        -------
+        numpy.ndarray
+            Boolean array where each element indicates whether the corresponding
+            chunk is valid (True) or faulty (False).
+        """
+        # Check if the camera has two gain channels
+        if outlier_mask.shape[1] == 2:
+            # Combine the outlier mask of both gain channels
+            outlier_mask = np.logical_or(
+                outlier_mask[:, 0, :],
+                outlier_mask[:, 1, :],
+            )
+        # Calculate the fraction of faulty pixels over the camera
+        faulty_pixels_percentage = (
+            np.count_nonzero(outlier_mask, axis=-1) / np.shape(outlier_mask)[-1]
+        ) * 100.0
```

Comment:
Do we really want to work in percentages here? I think just using fractions would be less confusing